src/video.c

```diff
@@ -112,11 +126,14 @@ static int video_compare_modes(const void *a, const void *b) {
 	return va->width * va->height - vb->width * vb->height;
 }
 
+static bool video_is_highdpi(void) {
+	return SDL_GetWindowFlags(video.window) & SDL_WINDOW_ALLOW_HIGHDPI;
+}
+
 void video_get_viewport_size(float *width, float *height) {
-	float w = video.current.width;
-	float h = video.current.height;
+	int w, h;
+	SDL_GL_GetDrawableSize(video.window, &w, &h);
 	float r = w / h;
```

Comment:
You changed the types of `w` and `h` to `int`, so `w / h` is now integer division, which is wrong. You have to cast one of the operands to `float` here.
```suggestion
	float r = (float)w / h;
```